components/layout/lib.rs

```diff
@@ -8,8 +8,11 @@
 #![feature(custom_derive)]
 #![feature(nonzero)]
 #![feature(plugin)]
+#![feature(proc_macro)]
 #![feature(raw)]
 #![feature(step_by)]
+#![feature(structural_match)]
+#![feature(rustc_attrs)]
```

Comment:
These are some features required by `serde_json`, but looks like they are no longer needed in `rustc-1.14.0`, I'll try removing them and see if it works.